packages/vuetify/src/labs/VPie/VPie.tsx

```diff
@@ -0,0 +1,378 @@
+            raw: item,
+          } as PieItem
+        })
+    })
+
+    const visibleItemsKeys = shallowRef<PieItem['key'][]>([])
+
+    watch(() => arcs.value.length, () => {
+      // reset when number of items changes
+      visibleItemsKeys.value = arcs.value.map(a => a.key)
+    }, { immediate: true })
+
+    const visibleItems = computed(() => {
+      // hidden items get (value: 0) to trigger disappearing animation
+      return arcs.value.map(item => {
+        return isActive(item)
+          ? item
+          : { ...item, value: 0 }
+      })
+    })
+
+    const total = computed(() => visibleItems.value.reduce((sum, item) => sum + item.value, 0))
+
+    const gaugeCut = toRef(() => Number(props.gaugeCut ?? 0))
+    const gaugeOffset = computed(() => (1 - Math.cos(Math.PI * Math.min(90, gaugeCut.value / 2) / 180)) / 2)
+    const rotateDeg = computed(() => `${gaugeCut.value ? (180 + gaugeCut.value / 2) : (props.rotate ?? 0)}deg`)
+
+    function arcOffset (index: number) {
+      return visibleItems.value
+        .slice(0, index)
+        .reduce((acc, s) => acc + (total.value > 0 ? s.value / total.value : 0) * (360 - gaugeCut.value), 0)
+    }
+
+    function arcSize (v: number) { return v / total.value * (100 - gaugeCut.value / 3.6) }
+
+    function colorFromPalette (index: number) {
+      if (props.palette.length === 0) return undefined
+      const paletteItem = props.palette[index % props.palette.length]
+      return typeof paletteItem === 'object' ? paletteItem.color : paletteItem
+    }
+
+    function patternFromPalette (index: number) {
+      if (props.palette.length === 0) return undefined
+      const paletteItem = props.palette[index % props.palette.length]
+      return typeof paletteItem === 'object' ? paletteItem.pattern : undefined
+    }
+
+    function isActive (item: PieItem) {
+      return visibleItemsKeys.value.includes(item.key)
+    }
+
+    function toggle (item: PieItem) {
+      if (isActive(item)) {
+        visibleItemsKeys.value = visibleItemsKeys.value.filter(x => x !== item.key)
+      } else {
+        visibleItemsKeys.value = [...visibleItemsKeys.value, item.key]
+      }
+    }
+
+    const tooltipItem = shallowRef<PieItem | null>(null)
+    const tooltipVisible = shallowRef(false)
+
+    let mouseLeaveTimeout = null! as ReturnType<typeof setTimeout>
+
+    function onMouseenter (item: PieItem) {
+      if (!props.tooltip) return
+
+      clearTimeout(mouseLeaveTimeout)
+      tooltipVisible.value = true
+      tooltipItem.value = item
+    }
+
+    function onMouseleave () {
+      if (!props.tooltip) return
+
+      clearTimeout(mouseLeaveTimeout)
+      mouseLeaveTimeout = setTimeout(() => {
+        tooltipVisible.value = false
+
+        // intentionally reusing timeout here
+        mouseLeaveTimeout = setTimeout(() => {
+          tooltipItem.value = null
+        }, 500)
+      }, 100)
+    }
+
+    return () => {
+      const itemProps = VPieSegment.filterProps(pick(props, [
```

Comment:
Why filterProps if they're already picked? There's only a few, why not explicit passthrough like you have for innerCut and hoverScale?